### DISCUSSION TITLE
Form old values

### DISCUSSION THREAD

**Discussion by DZ8540** [adonisjs/core#3485] [Help] [answered]
It would be nice to get the old user-entered values ​​from the form after the user has confirmed the form submission and got redirected back due to errors in the form.

In general, they can be obtained from flashMessages using the get() method with an empty string as the second argument. But there is a catch, if the user does not enter anything and sends the data, then after the redirect back, the field will contain null.

Example:
```blade
<input type="email" class="form-control" name="email" value="{{ flashMessages.get('email', '') }}" placeholder="hello@example.com">
```

To avoid this, you have to write like this:
```blade
<input type="email" class="form-control" name="email" value="{{ flashMessages.has('email') ? flashMessages.get('email') : '' }}" placeholder=" hello@example.com">
```

It is very uncomfortable. Is there any other way to get the old values?

**Comment by thetutlage** [comment, marked as answer]
Hey.

Instead of writing

```
flashMessages.has('email') ? flashMessages.get('email') : ''
```

You can write

```
flashMessages.get('email') || ''
```

**Reply by DZ8540** [reply]
Cool, thank you :)
